File: rdodo/R/change_vertical_speed.R
#' Send a "change vertical speed" command
#'
#' Request an aircraft to change vertical speed.
#'
#' @param aircraft_id
#' A string aircraft identifier
#' @param vertical_speed
#' The aircraft's new vertical speed in feet/min expressed as a non-negative
#' double or a quantity whose units can be converted to feet/min.
#'
#' @return
#' \code{TRUE} if successful. Otherwise \code{FALSE} and an error is thrown.
#'
#' @examples
#' \dontrun{
#' change_vertical_speed("test1234", vertical_speed = 10)
#' }
#' @import config httr
#' @export
change_vertical_speed <- function(aircraft_id, vertical_speed) {

  validate_aircraft_id(aircraft_id)
  validate_speed(vertical_speed)

  # Convert the given speed to feet/min
  units(vertical_speed) <- with(units::ud_units, ft/min)

  # TODO: replace string literals with config parameters from Bluebird.
  body <- list(
    "acid" = aircraft_id,
    "vspd" = as.double(vertical_speed)
  )
  post_call(endpoint = config_param("endpoint_change_vertical_speed"), body = body)
}

File: PyDodo/pydodo/change_vertical_speed.py

from . import settings
from . import utils

endpoint = settings.default['endpoint_change_vertical_speed']
url = utils.construct_endpoint_url(endpoint)

def change_vertical_speed(aircraft_id, vertical_speed):
    """
    Change aircraft speed, raise error if inputs are invalid or not successful
    """
    assert utils._check_string_input(aircraft_id), 'Invalid input {} for aircraft id'.format(aircraft_id)
    assert utils._check_speed(vertical_speed), 'Invalid input {} for vertical speed'

    json = {'acid': aircraft_id, 'vspd': speed}

    resp = requests.post(url, json=json)

    # if response is 4XX or 5XX, raise exception
    resp.raise_for_status()
    return True
